src/main/scala/org/dbpedia/spotlight/wikistats/JsonPediaParser.scala

```diff
@@ -59,5 +72,83 @@ class JsonPediaParser(lang:String)(implicit val sc: SparkContext,implicit val sq
   }
 
 
+  /*
+     Method to Get the list of Surface forms from the wiki
+   */
+  def getSfs() : RDD[String] = {
+
+    dfWikiRDD.select("wid","links.description")
+             .rdd
+             .map(artRow => (artRow.getList[String](1)))
+             .flatMap(sf => sf)
+  }
+
+  /*
+  Method to get the wid and article text from the wiki dump
+   */
+  //TODO Need to improve the filter condition of blank articles. we can include in the where clause of dataframe
+  def getArticleText(): RDD[(Long,String)] = {
+
+    dfWikiRDD.select("wid","wikiText")
+             .rdd
+             .map(artRow => {
+                 (artRow.getLong(0),artRow.getString(1))
+             })
+             .filter(artRow => artRow._2.length > 0)
+  }
+
+  /*
+   Logic to Get Surface Forms and URIs from the wikiDump
+   */
```

Comment:
can you add comments on what does each of the fields in the returning RDD stand for.
i.e: `(WikipediaID:Long, ...)`